### PR TITLE
EODHP-679 Workspace selection

### PR DESCRIPTION
**Pull request by james-hinton**
- Updated package json to fix actions
- Updated stylings for workspace ui page to a tab and card based approach
- Simple buttons to make fetch requests for tokens for S3 and DataHub API Key

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,34 +1,40 @@
...
-      case 'Credentials':
-        return <Credentials workspaceProject={workspaceProject} />;
-      case 'Workspaces':
-        return <div>Workspaces</div>;
-      default:
-        return <div>Default</div>;
-    }
-  };
+import TopBar from './components/TopBar/TopBar';
+import 'react-tabs/style/react-tabs.css';
```
Comment (marksmall) [inline src/App.tsx:5]: More an FYI, I tend to have CSS/SCSS files all imported after any JavaScript imports. I also think it is good to have them co-located, whereas the App style as at the top, then the tabs is in the middle. Just makes it more predictable and obvious, my opinion only.

I also notice we're not using **Absolute** imports, my own personal view is to use absolute imports for all but sibling components, then if we ever re-structure the app and then it just works.

File: src/components/TopBar/TopBar.scss [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  .workspace-selection {
+    display: flex;
+    flex-direction: column;
+    align-items: flex-end;
+    width: 100%;
+
+    .workspace-label {
+      font-size: 0.8rem;
+      color: #666;
+      font-weight: 300;
```
Comment (marksmall) [inline src/components/TopBar/TopBar.scss:28]: More something we should look at in the future. I don't use Stylelint much, but thought styles would be ordered, unless the order somehow impacted the display. So I expected `font-size` and `font-weight` to be together, but having looked at this in VS Code and run stylelint manually against this branch, it doesn't. There is a [stylelint-order](https://www.npmjs.com/package/stylelint-order) that we could add, I'll look to do this in the starter project and get back to you on this. Not looking for anything to be done right now.

File: src/components/TopBar/TopBar.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          })}
+          onChange={(selectedOption) => setActiveWorkspace(selectedOption?.value)}
+        />
+      </div>
+
+      <div>
+        <p className="heading">Your Workspace</p>
+        <p className="date">
+          {activeWorkspace} -{' '}
+          {new Date().toLocaleDateString('en-US', {
```
Comment (marksmall) [inline src/components/TopBar/TopBar.tsx:37]: Is there a reason we are using the US Date? I normally have battles on this as I prefer iso 8601 e.g. 2024-09-18 as opposed to the constant battles I've had over US/UK dates. The world seems pretty split between the 3 formats, but only iso 8601 I feel is unambiguous. Again, not asking for change, more a question really.

File: src/components/TopBar/TopBar.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            month: 'long',
+            year: 'numeric',
+          })}
+        </p>
+      </div>
+    </div>
+  );
+};
+
+export default TopBar;
```
Comment (marksmall) [inline src/components/TopBar/TopBar.tsx:49]: I tend to prefer `named exports` so usage around an app is consistent, I only use defaults when forced to. I personally don't find it useful to have written a component, then have it used in multiple places all using different import names, make my brain have to work harder. What do you think?

File: src/pages/Applications/components/Application.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      </div>
+      <div className="app-details">
+        <h3>{name}</h3>
+        <p>{description}</p>
+      </div>
+    </button>
+  );
+};
+
+export default Application;
```
Comment (marksmall) [inline src/pages/Applications/components/Application.tsx:29]: Same here on the `named` vs `default` exports, also think `React.FC` is generally falling by the wayside these days and that components are now being written as:

```JavaScript
export const Application = ({ name, description, icon: Icon }: ApplicationProps) ={...}
```
Here is a reddit article on this https://www.reddit.com/r/reactjs/comments/wjq51d/is_reactfc_not_recommended_what_are_other/, I also googled `should I use React.FC` and got a number of pros/cons, but one that caught my eye was PR in create-react-app, by facebook themselves stating `React-FC` is unecessary https://github.com/facebook/create-react-app/pull/8177. I don't think it is a big deal to stop this PR getting merged, but I suggest you no longer use `React.FC` for reasons given in links above.

**Review by marksmall — COMMENTED** [review]
Made a few FYI comments, nothing major. It would be nice to use `absolute` imports as there is linking to ensure they are ordered correctly, which isn't applied to relative imports.

Let me know what you think as I think we could get this merged pretty soon, depending on what changes in response to comments made you want to make.

File: src/components/TopBar/TopBar.tsx (hunk 4 above)
Comment (james-hinton) [reply src/components/TopBar/TopBar.tsx:49]: Ah, muscle memory with export default, I have updated them

File: src/App.tsx (hunk 1 above)
Comment (james-hinton) [reply src/App.tsx:5]: Updated the imports

File: src/pages/Applications/components/Application.tsx (hunk 5 above)
Comment (james-hinton) [reply src/pages/Applications/components/Application.tsx:29]: Thanks for the link, I've updated this

**Pull request merged**